packages/components/src/components/button/_button.scss

```diff
@@ -118,9 +118,9 @@
     @include button-theme(
       transparent,
       transparent,
-      $interactive-04,
+      $icon-01,
```

Comment:
Also I wrote the wrong token in the initial [comment](https://github.com/carbon-design-system/carbon/issues/3190#issuecomment-506765800). Ghost button should be using `$link-01` for text + icon (not `$icon-01`). My bad guys.